### PR TITLE
Add compose_ids and signing_intent options

### PR DESCRIPTION
**Pull request by bfontecc007**
Allow --compose-id and --signing-intent parameters in koji-containerbuild cli. Pass compose_ids and signing_intent to osbs.

### CODE REVIEW

File: tests/test_kcb.py [hunk 1]
```diff
@@ -701,6 +724,7 @@ def test_cli_args(self, tmpdir, scratch, wait, quiet,
             assert build_opts.module == module
         else:
             assert build_opts.release == release
+        assert build_opts.compose_ids == compose_ids
```
Comment (lcarva) [inline tests/test_kcb.py]: `assert build_opts.signing_intent == signing_intent` ?

File: tests/test_kcb.py [hunk 2]
```diff
@@ -781,6 +805,57 @@ def test_isolated_scratch_restriction(self, tmpdir, scratch, isolated, valid):
         assert parsed_args == expected_args
         assert opts == expected_opts
 
+    @pytest.mark.parametrize(('additional_args', 'raises'), (
+        ({'compose_ids': None, 'signing_intent': None}, False),
```
Comment (lcarva) [inline tests/test_kcb.py]: Could omit the keys that have `None` value.

File: tests/test_kcb.py (hunk 1 above)
Comment (bfontecc007) [reply tests/test_kcb.py]: added.

File: tests/test_kcb.py (hunk 2 above)
Comment (bfontecc007) [reply tests/test_kcb.py]: omitted.

Comment (lcarva) [reply tests/test_kcb.py]: err.. I meant omit the key, not the scenario:
```
{},
{'compose_ids': [1, 2]},
...
```

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 3]
```diff
@@ -506,6 +510,9 @@ def createContainer(self, src=None, target_info=None, arches=None,
         if not arches:
             raise ContainerError("arches aren't specified")
 
+        if signing_intent and compose_ids:
+            raise ContainerError("signing_intent used with compose_ids")
```
Comment (twaugh) [inline koji_containerbuild/plugins/builder_containerbuild.py]: This doesn't seem like the right exception to raise. Probably `koji.BuildError` is better for this. I see `ContainerError` is also raised if arches is not specified -- that also looks wrong to me.

File: tests/test_kcb.py (hunk 2 above)
Comment (bfontecc007) [reply tests/test_kcb.py]: oh I had no idea you could do that

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 3 above)
Comment (lcarva) [reply koji_containerbuild/plugins/builder_containerbuild.py]: Good catch! It looks like this is also how it's done in RPM land:

```python
def getArchList(self, build_tag, h, extra=None):                                  
    # get list of arches to build for                                             
    buildconfig = self.session.getBuildConfig(build_tag, event=self.event_id)     
    arches = buildconfig['arches']                                                
    if not arches:                                                                
        #XXX - need to handle this better                                         
        raise koji.BuildError("No arches for tag %(name)s [%(id)s]" % buildconfig)
    tag_archlist = [koji.canonArch(a) for a in arches.split()]                    
    self.logger.debug('arches: %s' % arches)                                      
    if extra:                                                                     
        self.logger.debug('Got extra arches: %s' % extra)                         
        arches = "%s %s" % (arches,extra)                                         
    archlist = arches.split()                                                     
```
https://pagure.io/koji/blob/master/f/builder/kojid#_958

File: tests/test_kcb.py [hunk 4]
```diff
@@ -781,6 +806,57 @@ def test_isolated_scratch_restriction(self, tmpdir, scratch, isolated, valid):
         assert parsed_args == expected_args
         assert opts == expected_opts
 
+    @pytest.mark.parametrize(('additional_args', 'raises'), (
+        ({}, False),
+        ({'compose_ids': [1, 2], 'signing_intent': None}, False),
```
Comment (lcarva) [inline tests/test_kcb.py]: Let's also remove all keys that are set to `None`. In full this should be:
```python
({}, False),
({'compose_ids': [1, 2],}, False),
({'signing_intent': 'intent1'}, False),
({'compose_ids': [1, 2], 'signing_intent': 'intent1'}, True)
```

**Comment by lcarva** [comment]
Updated release notes draft:
> - `compose_ids` and `signing_intent` task options

**Pull request merged**